contracts/facets/TermFacet.sol

```diff
@@ -126,7 +130,17 @@ contract TermFacet is ITerm {
 
         require(!collateral.isCollateralMember[msg.sender], "Reentry");
 
-        uint memberIndex = collateral.counterMembers;
+        uint memberIndex;
+
+        for (uint i; i < term.totalParticipants; ) {
+            if (collateral.depositors[i] == address(0)) {
+                memberIndex = i;
+                break;
+            }
+            unchecked {
+                ++i;
+            }
+        }
```

Comment:
From this point on, you could call `_joinTerm(uint _termId, bool _optYield, uint _position)`.

What `_joinTerm(uint _termId, bool _optYield)` would do is get the next available position, then call `_joinTerm(uint _termId, bool _optYield, uint _position)`.

Other than this it's fine IMO

Reply:
Like this idea. let me commit that

Reply:
Commit https://github.com/TakafulDAO/takaturn/commit/7dffd34c8c380bfce14200859752e3ca2285b058 and tests on commit https://github.com/TakafulDAO/takaturn/commit/810b5a8277e56c40289350c97dcc7c5929415a70